crates/telio-firewall/src/firewall.rs

```diff
@@ -950,6 +989,82 @@ impl Firewall for StatefullFirewall {
             }
         }
     }
+
+    fn reset_connections(
+        &self,
+        sink4: &mut dyn io::Write,
+        _: &mut dyn io::Write,
+    ) -> io::Result<()> {
+        telio_log_debug!("Constructing connetion reset packets");
+
+        if let Ok(mut tcp) = self.tcp.lock() {
+            telio_log_debug!("TCP connections to reset: {}", tcp.len());
+
+            const IPV4_LEN: usize = 20;
+            const TCP_LEN: usize = 20;
+
+            let mut ipv4buf = [0u8; IPV4_LEN];
+            let mut tcpbuf = [0u8; TCP_LEN];
+
+            // Write most of the fields for TCP and IP packets upfront
+            #[allow(clippy::expect_used)]
+            let mut tcppkg =
+                MutableTcpPacket::new(&mut tcpbuf).expect("TCP buffer should not be too small");
+            tcppkg.set_flags(TcpFlags::RST);
+            tcppkg.set_data_offset(5);
+
+            #[allow(clippy::expect_used)]
+            let mut ipv4pkg =
+                MutableIpv4Packet::new(&mut ipv4buf).expect("IPv4 buffer should not be too small");
+            ipv4pkg.set_version(4);
+            ipv4pkg.set_header_length(5);
+            ipv4pkg.set_total_length((IPV4_LEN + TCP_LEN) as _);
+            ipv4pkg.set_flags(Ipv4Flags::DontFragment);
+            ipv4pkg.set_ttl(0xFF);
+            ipv4pkg.set_next_level_protocol(IpNextHeaderProtocols::Tcp);
+
+            let mut ipv4pkgbuf = [0u8; IPV4_LEN + TCP_LEN];
+            for (key, val) in tcp.iter() {
```

Comment:
I realized that we missed one thing in the RFC LLT-0042, when killing the connections, we should not be killing all of the connections, but only the ones which were created over `exit_node` peer.

The clearest manifestation of this would be that when user is using meshnet and VPN at the same time, reconnecting from one VPN server to another - should not kill meshnet connections. IMHO, attaching public key of the peer to the connection state or event implementing per-peer conntrack as suggested in LLT-4502 would resolve this issue naturally. But when killing the connections, peer's public key will have to be passed all the way from src/device/mod.rs :/

Reply:
Does this work other way around as well? Like when user starts routing traffic through another node, it should not kill VPN connections I suppose as well. If `exit_node` is reused for meshnet and for VPN I think the problem is solved in one place

Reply:
Generally routing through meshnet exit node and routing through VPN server has the same problem, so the resolution would be the same. 

You want to kill connections which has been established via the old exit node (whether it is VPN or meshnet exit node) and to avoid touching connections to/via other peers.

Reply:
This diagram shows "more full" picture.

We want to kill only dashed connections:
![Stale-TCP-conns-when-VPN+meshnet](https://github.com/NordSecurity/libtelio/assets/532330/ca268c4c-f7da-40b5-9a33-76828a00649e)

Reply:
@Jauler I've pushed the relevant changes, only for TCP though as this is the scope of the PR